include/meshio/vectors.hpp

```diff
@@ -52,41 +93,85 @@ class Vec2 {
     }
 
     bool operator==(const Vec2& other) {
-        return ((x == other.x) && (y == other.y));
+        return ((elements[0] == other.X()) && (elements[1] == other.Y()));
     }
+
+  private:
+    T elements[2];
 };
 
 template<class T>
-class Vec3 {
+class Vec3 : public VecBase<T> {
   public:
-    T x,y,z;
+    typedef T *iterator;
+    typedef const T *const_iterator;
+
+    Vec3(T pX, T pY, T pZ) : elements{pX, pY, pZ} {}
+    Vec3() : Vec3({static_cast<T>(0), static_cast<T>(0), static_cast<T>(0)}) {}
```

Comment:
so, I didn't say it right... in the Travis build, servers are testing on mac and linux instances... I referred mac instance as mac worker, I am pretty sure I am wrong, since you didn't understand what I said :P